docs/modules/install/pages/customization.adoc

```diff
@@ -0,0 +1,85 @@
+# Change logic in decdidim-core #
+
+There are occasions you find yourself in a situation that you need to change Decidim core logic.  There are two ways to accomplish this goal, and we are going through these options here.
+
+## 1. Direct change from your application ##
+
+This method is more straight-forward, and we discuss it here first. To change the logic in decidim, identify which part of the source code you need to change. The best way to achieve this is to identify the module being responsible for handling the task. Next, create the same path on our project, and copy the file to the same destination in our project. Finally, change the code as you wish and make sure that the changes are being applied.
+Lets have an example to make this clear. Lets add a rule to the Admin panel, where Admin can not add projects with budgets more than half the total budget. Once we applied such limitation, admin should not be able to add a proposal with the cost of 5,001 Euros to a budget with total of 10,000 Euros.
+
+We want to add a validation rule to achieve this goal. For that to happen, we go through the following steps:
+
+. First, we need to investigate the original source code, where the validation for new projects being generated(in this case, it is decidim-budgets/App/form/admin/create_proposal).
+. Now that we find the original source code, we need to over-ride this file. For that to happen, we create the following destination(if it has not already existed), and copy all contents of the original files there(in this particular example, change the validation for :budget_amount.
+
+NOTE: Please note the version of decidim on github, since it may not be in sync with your local version.
+
+[start=3]
+. Next, in copied file, lets add/edit a validation for maximum allowable value:
+
+. save the file and re-run the server from the console, and check if the changes has taken effect.
+
+[#img-sunset]
+. Trying to create a proposal with the budget of €5,001 for a budget with the total amount of €10,000(My budget) reults in error(buttom-right error message).
+
+image::maximum_budget.png[Maximum budget error]
+
+## 2. Over-ride logic with concern ##
+
+Concerns allow us to include modules with methods and constants into a class so that the including class can use them. After identifying the original code to be modified, we need to first create a concern. Put it simple, concern means we highlight only the parts of the original code to be changes(This might be a block of a code we need to change, or add/edit an existing method, etc). There is a convention where to put concerns, which is inside a folder which we call it concern. Take the previous example, in which we create a folder named concern inside form folder. Finally, you should make that file available prior to applying changes.
```

Comment:
```suggestion
Concerns allow us to add functionality into existing classes so that the including class can use the added functionality. This is a pattern provided by the Rails framework and you can learn more about it from the https://api.rubyonrails.org/classes/ActiveSupport/Concern.html[Rails API documentation]

Simply put, a concern allows developers to change only the parts of the original code that need to be changed. This might be a block of a code we need to change, or add/edit an existing method, etc.

The benefit of using concerns is that when we update Decidim gems to new versions, we do not have to necessarily go through all our customizations and we can just focus on those parts that we really want to change. This is slightly harder to do than the first example but in the long run it will pay you back as you will save a lot of time when updating Decidim.

Concerns should be placed into the `concerns` folder within the top-level folder where you want to apply these concerns. In this case, we are adding a concern for the forms classes, so we should place the concerns related to forms to `app/forms/concerns`. If you do not have this folder available, create it first `mkdir -p app/forms/concerns`.
```